### PR TITLE
[arellano] get rid of jitclass

### LINKED ISSUE #129: [arellano] use namedtuple
Remove class and use namedtuple in lecture [arellano](https://jax.quantecon.org/arellano.html).

### PR DESCRIPTION
**Pull request by shlff**
Hi @jstac this PR gets rid of class and uses namedtuple, as mentioned in #129 .

### CODE REVIEW

**Comment by jstac** [comment]
@shlff , this is labeled as "in-work".  Is it ready to review?

**Comment by shlff** [comment]
Thanks @jstac . I've been reviewing this PR, and I will get back to you very soon.

**Comment by shlff** [comment]
Thanks @jstac . This PR is ready to review. Here is a preview:

https://65e521825ca446fa847b6ccd--incomparable-parfait-2417f8.netlify.app/arellano

In addition to using namedtuple, I also modified the words previously mentioning the class and corrected some typos.

**Comment by jstac** [comment]
Thanks @shlff , appreciated.

@HumphreyYang , can you please review?

File: lectures/arellano.md [hunk 1]
```diff
@@ -365,55 +366,43 @@ The output process is discretized using a [quadrature method due to Tauchen](htt
...
+                    B_min=-0.45,   # Smallest B value
+                    B_max=0.45,    # Largest B value
+                    y_size=51,     # Grid size for income
+                    β=0.953,            # Time discount parameter
+                    γ=2.0,              # Utility parameter
+                    r=0.017,            # Lending rate
+                    ρ=0.945,            # Persistence in the income process
+                    η=0.025,            # Standard deviation of the income process
+                    θ=0.282,            # Prob of re-entering financial markets
+                    def_y_param=0.969): # Parameter governing income in default
```
Comment (HumphreyYang) [inline lectures/arellano.md:389]: Move the comments on parameters to where we define the `namedtuple`.

**Review by HumphreyYang — COMMENTED** [review]
Many thanks @shlff for the careful changes. Please find my minor comments above.

File: lectures/arellano.md (hunk 1 above)
Comment (shlff) [reply lectures/arellano.md:389]: Thanks @HumphreyYang , I will add the comments to both places, given the comments from another PR.

File: lectures/arellano.md [hunk 2]
```diff
@@ -365,55 +366,43 @@ The output process is discretized using a [quadrature method due to Tauchen](htt
 
 As we have in other places, we accelerate our code using Numba.
 
-We define a class that will store parameters, grids and transition
+We define a namedtuple to store parameters, grids and transition
 probabilities.
 
 ```{code-cell} ipython3
-:hide-output: false
+Arellano_Economy = namedtuple('Arellano_Economy', ('β', 'γ', 'r', 'ρ', 'η', 'θ', \
```
[the remark this answers is not in the document]
Comment (shlff) [reply lectures/arellano.md:373]: Good idea! We should change the name in the original lecture too:

https://python-advanced.quantecon.org/arellano.html

File: lectures/arellano.md [hunk 3]
```diff
@@ -695,18 +687,15 @@ def simulate(model, T, v_c, v_d, q, B_star, key):
...
 
 Details on how to compute the figures are reported as solutions to the
 exercises.
 
 The first figure shows the bond price schedule and replicates Figure 3 of
-Arellano, where $ y_L $ and $ Y_H $ are particular below average and above average
+{cite}`Are08`, where $ y_L $ and $ Y_H $ are particular below average and above average
 values of output $ y $.
 
 ![https://python-advanced.quantecon.org/_static/lecture_specific/arellano/arellano_bond_prices.png](https://python-advanced.quantecon.org/_static/lecture_specific/arellano/arellano_bond_prices.png)
```
[the remark this answers is not in the document]
Comment (shlff) [reply lectures/arellano.md:701]: Thanks @HumphreyYang . I reckon @mmcky have already done that in the PR:

https://github.com/QuantEcon/lecture-jax/pull/141

**Comment by HumphreyYang** [comment]
[celebrate]     Humphrey Yang reacted to your message:
________________________________

**Comment by HumphreyYang** [comment]
[like]  Humphrey Yang reacted to your message:
________________________________

**Comment by HumphreyYang** [comment]
[0]     Humphrey Yang reacted to your message:
________________________________

**Comment by mmcky** [comment]
@shlff do you know why you needed to use a force push. They are pretty dangerous and I encourage you to avoid using them. Would you like to update this PR from the `main` as I fixed the image issues (as you commented above).

**Comment by shlff** [comment]
Sorry @mmcky I realised the risk of force push and will not use it anymore. 

Yep it would be great if we can update this PR from the main. I reckon that after that some check failure should be gone.

**Comment by shlff** [comment]
Thanks @HumphreyYang @jstac and @mmcky . I resolved the issues mentioned in the comments above.

This PR is ready to review again and here is a preview:

https://65e64dd91341d614052e79ae--incomparable-parfait-2417f8.netlify.app/

**Comment by jstac** [comment]
Many thanks @shlff 

- Camel case has no spaces, so Arellano_Economy -> ArellanoEconomy
- PEP8 should apply as much as possible -- in particular, max 80 characters per line
- The following lines are unnecessary because the arrays are already JAX arrays

```python
    B_grid = jax.device_put(B_grid)
    y_grid = jax.device_put(y_grid)
```

**Comment by shlff** [comment]
Thanks @jstac . 

It seems that the commit with changes I made in the morning has not been pushed to this PR.

I will make the changes suggested by your comments along with those changes.

**Comment by shlff** [comment]
Thanks @jstac . The following tasks have been done:
- [x] Camel case has no spaces, so Arellano_Economy -> ArellanoEconomy
- [x] PEP8 should apply as much as possible -- in particular, max 80 characters per line
- [x] The following lines are unnecessary because the arrays are already JAX arrays

This PR is ready to review again, and please find a preview:

https://65e6b96ed004475661581086--incomparable-parfait-2417f8.netlify.app/arellano

**Comment by jstac** [comment]
Hi @shlff , the first code block (ArellanoEconomy = ...) has lines more than 80 characters.  Same with the next one.

**Comment by shlff** [comment]
Good catch @jstac . I modified all the code cells, which should all strictly follow the max 80 characters rule except for line 601.

Please find the preview above.

**Comment by jstac** [comment]
I'm confused @shlff .  In that preview there are multiple lines that exceed 80 characters, even in the first code block.

**Comment by jstac** [comment]
That's strange.  Try counting the characters in this line:

```                              'P',       # Markov matrix governing the income process
```

**Comment by shlff** [comment]
Thanks @jstac . I figured out:

I guess you reviewed it using an old preview. 

If you review the same line from the most recent preview below, then this line should be 80 characters:

https://65e6c6b323165e623b58b2dd--incomparable-parfait-2417f8.netlify.app/arellano

**Comment by jstac** [comment]
Thanks @shlff 

I've readjusted those lines so the reader doesn't need to scroll and the comments are aligned.

**Pull request merged**